origin-growth/src/rules/rules.js

```diff
@@ -1,11 +1,59 @@
 const Sequelize = require('sequelize')
+const BigNumber = require('bignumber.js')
 
 const db = require('../models')
-const { GrowthEventTypes, GrowthEventStatuses } = require('../enums')
+const {
+  GrowthEventTypes,
+  GrowthEventStatuses,
+  GrowthCampaignStatuses,
+  GrowthActionStatus,
+  GrowthActionType } = require('../enums')
 
 // System cap for number of rewards per rule.
 const MAX_NUM_REWARDS_PER_RULE = 1000
 
+const sumUpRewards = (rewards) => {
+  if (rewards === null || rewards.length === 0){
+    return null
+  }
+
+  const finalReward = rewards.reduce((first, second) => {
+    if (first.currency !== second.currency)
+      throw new Error(`At least two rewards have different currencies. ${first.currency} ${second.currency}`)
+    return {
+      amount: BigNumber(first.amount).plus(BigNumber(second.amount)),
+      currency: first.currency
+    }
+  })
+
+  return {
+    amount: finalReward.amount.toString(),
+    currency: finalReward.currency
+  }
+}
+
+const eventTypeToActionType = (eventType) => {
+  if (eventType === 'ProfilePublished'){
```

Comment:
alternative to this long list of if/else statements: do a lookup in a dictionary.

eventTypeToActionTypeMap = {
 ProfilePublished: 'profile',
 EmailAttestationPublished: 'email'
 ...
}
return eventTypeToActionTypeMap[eventType]

Reply:
Nice, that is much cleaner... 👍